### DISCUSSION TITLE
Conditional insert

### DISCUSSION THREAD

**Discussion by well1791** [prestodb/presto#18706] [Q&A] [answered]
What would be the presto solution to a problem like [this](https://stackoverflow.com/questions/16636698/do-conditional-insert-with-sql)

I have a python script that does this essentially

```
            f"INSERT INTO {self.table} ({fields}) VALUES ({values})"
            f"SELECT t1.id FROM {self.table} t1 "
            f"WHERE NOT EXISTS(SELECT t2.id FROM {self.table} t2 WHERE t2.id = t1.id)"
```

The idea is to insert only if there is no row that matches the same field-values

or, is it possible to run only one statement to achieve the same result?

**Comment by bitsondatadev** [comment, marked as answer]
Hey @well1791, looks like you're looking for a feature like [`MERGE`](https://trino.io/docs/current/sql/merge.html). This isn't supported in Presto yet, but take a look at Trino (a fork off Presto by its original creators).
